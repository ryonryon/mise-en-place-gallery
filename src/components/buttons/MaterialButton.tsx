import styled from "@emotion/styled";
import * as React from "react";
import { secondary, white } from "../../constants/colors";

const MaterialStyledButton = styled.button`
  background-color: ${white};
  border-radius: 5px;
  border: none;
  color: ${secondary};
  height: 30px;
  width: 150px;
  cursor: pointer;
  margin: 16px;
  box-shadow: 0px 8px 15px rgba(0, 0, 0, 0.1);
  &:hover {
    box-shadow: 2px 10px 15px rgba(0, 0, 0, 0.3);
  }
`;

interface Props extends React.ButtonHTMLAttributes<HTMLButtonElement> {}

const MaterialButton = ({ ...rest }: Props) => <MaterialStyledButton {...rest}></MaterialStyledButton>;

export default MaterialButton;
