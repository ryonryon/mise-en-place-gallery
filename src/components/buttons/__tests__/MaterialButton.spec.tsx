import * as React from "react";
import * as renderer from "react-test-renderer";

import MaterialButton from "../MaterialButton";

describe("MaterialButton test", () => {
  it("tests MaterialButton renders correctly ", () => {
    const materialButton = renderer.create(<MaterialButton />).toJSON();
    expect(materialButton).toMatchSnapshot();
  });
});
