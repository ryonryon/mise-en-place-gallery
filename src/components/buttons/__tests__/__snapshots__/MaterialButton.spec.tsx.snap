// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MaterialButton test tests MaterialButton renders correctly  1`] = `
<button
  className="css-1uurvcs"
/>
`;
